<launch>
    <arg name="output"            default="screen"    doc="display output to screen or log file"/>
    <arg name="TrajTopic"   default="/planning/trajectory"    doc="ROS topic for the reference trajectory"/>
    <arg name="PoseTopic"   default="/vicon/rc_car_rear/rc_car_rear"    doc="ROS topic for the pose"/>
    <arg name="ControllerTopic"   default="/control/rc_control"    doc="ROS topic to publish"/>
    <arg name="Horizon"         default="0.25"              doc="Frequency to update the controller" />
    <arg name="Step"      default="5"              doc="Frequency to update the controller" />
    <node pkg="traj_tracking_ros" type="traj_tracking_node.py" name="traj_tracking_node" output="$(arg output)" args="" required="true">
        <param name="TrajTopic"           value="$(arg TrajTopic)" />
        <param name="PoseTopic"           value="$(arg PoseTopic)" />
        <param name="ControllerTopic"           value="$(arg ControllerTopic)" />
        <param name="TrackHorizon"           value="$(arg Horizon)" />
        <param name="TrackStep"           value="$(arg Step)" />
        <param name="TrackParamsFile"     value="$(find traj_tracking_ros)/params/modelparams.yaml" />
        <param name="TrackBasePath"     value="$(find traj_tracking_ros)" />
    </node>
</launch>
